docs/DeveloperGuide.md

```diff
@@ -6,24 +6,177 @@
+- `Timetable`: Stores the list of courses
+- `GradeChecker`: Provides access to the grades of the given period
+
+Below is the class diagram to illustrate this:
+![Grades_ClassDiagram](uml/Grades_ClassDiagram-0.png)
+
+<br>
+
+When `addGrade()` is called, it finds the course according to the course code. Then, the grade will be set to that course.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddGrade_SequenceDiagram](uml/AddGrade_SequenceDiagram-0.png)
+
+<br>
+
+When `removeGrade()` is called, it finds the course according to the course code. Then, the `lettergrade` of that course is set to `null`.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![RemoveGrade_SequenceDiagram](uml/RemoveGrade_SequenceDiagram-0.png)
+
+
+The grades keyed in by the user are **String** type, e.g.`"A"`.
+In the `Grade` class, the `setNumberGrade()` method converts the grade into **double** type, which is more appropriate for the calculation of GPA and total grades.
+
+There are 3 overloaded variations of the `checkGrade()` method in the `GradeChecker` class.
+The first only takes in Timetable `timetable`, which records the current plan of the user.
+The second takes in an additional integer `year`, while the third takes in two additional integers `year` and `term`.
+
+### Ui Component
+The Ui class is implemented with static methods to facilitate easy access to user interface functionalities without 
+needing to instantiate the class. It manages a variety of messages that inform the user about the outcome of their commands, guide them on the correct command syntax, and display error messages for incorrect inputs.
+
+**Key Methods and Their Design Choices:**
+
+- `printLogo()`: Displays the PlaNus logo at startup. This method enhances brand recognition and provides a welcoming 
+user interface.
+
+- `printHelp()`: Lists all available commands and their syntax. This method is crucial for aiding new users in 
+familiarizing themselves with the application's functionalities.
+
+- `printErrorMessage(String message)`: Accepts a string parameter and displays it as an error message. This method 
+is designed to provide specific feedback on what went wrong, enhancing the user's ability to correct errors.
+
+- `getUserCommand()`: Reads the user's input command. Implemented using a Scanner object, this method bridges the user 
+command input with the application's processing logic.
+
+#### Integration with Other Components
+
+The Ui component works closely with the `Parser`, `Storage`, and various command classes. It acts as the output medium 
+for messages generated by these components and reads input that is subsequently parsed and acted upon.
+
+*   **Parser Component**: The Ui displays help and error messages based on parsing results.
+
+*   **Storage Component**: Notifications about file operations (e.g., file creation success or failure) are relayed to 
+the user via the Ui.
+
+![Ui_ClassDiagram](uml/UiClassDiagram.png)
```

Comment:
There's clear delineation of the UI component's key methods and their purposes, which helps in understanding how user interactions are handled.